MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm

```diff
@@ -1177,17 +1204,18 @@ static VkFormat mvkFormatFromOutput(const SPIRVShaderOutput& output) {
 			// If this is the case, fetch a translated artificial buffer binding, using the same MTLBuffer,
 			// but that is translated so that the reduced VA offset fits into the binding stride.
 			const VkVertexInputBindingDescription* pVKVB = pVI->pVertexBindingDescriptions;
+			uint32_t attrSize = 0;
 			for (uint32_t j = 0; j < vbCnt; j++, pVKVB++) {
 				if (pVKVB->binding == pVKVA->binding) {
-					uint32_t attrSize = getPixelFormats()->getBytesPerBlock(pVKVA->format);
+					attrSize = getPixelFormats()->getBytesPerBlock(pVKVA->format);
 					if (pVKVB->stride == 0) {
 						// The step is set to constant, but we need to change stride to be non-zero for metal.
 						// Look for the maximum offset + size to set as the stride.
 						uint32_t vbIdx = getMetalBufferIndexForVertexAttributeBinding(pVKVB->binding);
 						auto vbDesc = inputDesc.layouts[vbIdx];
 						uint32_t strideLowBound = vaOffset + attrSize;
 						if (vbDesc.stride < strideLowBound) vbDesc.stride = strideLowBound;
-					} else if (vaOffset + attrSize > pVKVB->stride) {
+					} else if (vaOffset && vaOffset + attrSize > pVKVB->stride) {
```

Comment:
Ah! Good. When I was reviewing this PR earlier...I noticed that `attrSize > pVKVB->stride` would trigger an alternate buffer to be created and was going to ask you about that. Glad you caught it too and updated it.